packages/fxa-auth-db-mysql/db-server/test/backend/db_tests.js

```diff
@@ -2597,6 +2597,84 @@ module.exports = function (config, DB) {
       })
     })
 
+    describe('account subscriptions', () => {
+      const subs = [...Array(10)].map(() => 'sub-' + Math.random())
+      let account
+
+      beforeEach(() => {
+        account = createAccount()
+        return db.createAccount(account.uid, account)
+      })
+
+      it('should create a new subscription for account', () => {
+        return db.createAccountSubscription(account.uid, subs[0], 'prod0', Date.now())
+          .then(result => assert.deepEqual(result, {}))
+      })
+
+      it('should fail to create for unknown account', () => {
+        return db.createAccountSubscription('12312312312', subs[1], 'prod2', Date.now())
+          .then(assert.fail, err => assert.equal(err.errno, 116, 'not found'))
+      })
+
+      it('should fail to create duplicate subscription id', () => {
+        const account2 = createAccount()
+        return db.createAccount(account2.uid, account2)
```

Comment:
same as a comment below, could/should these new tests use `async /await`?

Reply:
I could switch to async / await, but I was trying to follow existing patterns. If switching is a thing we want to do, I can rework things

Reply:
Reworked to use async / await, which also drops the implicit returns from arrow functions